wandb/sdk/launch/agent/agent.py

```diff
@@ -53,9 +125,11 @@ def __init__(self, api: Api, config: Dict[str, Any]):
         self._project = config.get("project")
         self._api = api
         self._base_url = self._api.settings().get("base_url")
-        self._jobs: Dict[Union[int, str], AbstractRun] = {}
         self._ticks = 0
-        self._running = 0
+        manager = Manager()
+        self._jobs = manager.dict()
+        _lock = manager.Lock()
+        self._pool = Pool(initializer=init_pool_processes, initargs=(self._jobs, _lock))
```

Comment:
Okay this looks sick, but can I make another thread pool for schedulers? I want to have another one of these that is identical, just manages scheduler jobs separately.... No conflicts here?

Reply:
You'll want to use the same pool. You can make another `apply_async()` call for your own use case, and will likely want more globals to track them - the same way I have `_jobs` and `_lock`... I'll probably want to rename `_lock` if there are going to be multiple locks in the future